codeandcoffeelb/codeandcoffeelb.github.io
# Feature/#71 add vagrantfile

Apologies for the merge conflicts.

This branch adds a Vagrantfile and shell-based vagrant provisioning, to allow easier local testing for those who struggle with local installs of Ruby. Implementing this necessitated a variety of changes:

1. Upgrading to Jekyll 3 (via the github-pages gem)
2. Per Jekyll 3, moving source code into a subdirectory to avoid conflicts. This added to new properties to _config.yml: `source` and `destination`, and moved nearly every file in the repo under the new `_source` directory.

How to test/use:

1. from the repo root, run `vagrant destroy` and then `vagrant up`
2. Watch the provisioning fly by. Sorry, this may take a few minutes, particularly if you haven't used vagrant before or don't have a local cache of the `ubuntu/trusty64` image.
3. See that it completes without error and note the instructions at the end.
4. Connect via ssh by running `vagrant ssh`
5. `cd /vagrant`
6. `bundle exec jekyll serve --host 0.0.0.0 --port 8124`
7. On your host machine, navigate to `http://0.0.0.0:8124` in your browser
8. Profit?

Notes:
1. This uses rbenv to install an appropriate version of Ruby (2.2.3)
2. This uses bundler to install all Gems in the Gemfile, including Jekyll itself
3. Meh.
